programs/Makefile

```diff
@@ -39,9 +39,15 @@ RELEASE?= r131
 DESTDIR?=
 PREFIX ?= /usr/local
 CFLAGS ?= -O3
-CFLAGS += -std=c99 -Wall -Wextra -Wundef -Wshadow -Wcast-qual -Wcast-align -Wstrict-prototypes -pedantic -DLZ4_VERSION=\"$(RELEASE)\"
-FLAGS  := -I../lib $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
+CFLAGS += -std=c99 -Wall   -Wextra -Wundef -Wshadow -Wcast-qual -Wcast-align -Wstrict-prototypes -pedantic -DLZ4_VERSION=\"$(RELEASE)\"
+
+
+ifeq ($(shell uname --hardware-platform), i386)
+           CFLAGS += -march=prescott
```

Comment:
This is hardware specific.

Can you prove that `-march=prescott` will always be beneficial when `uname --hardware-platform` returns `i386` ?
What about AMD platforms ? What about `clang` ?

Sure enough, there are _some_ situations in which it will prove better, like probably for your platform.
That being said, setting this parameter as "default" (for i386) suppose it will be beneficial for the vast majority of systems concerned, a fairly strong claim that would deserve to be backed by good evidences.

Note that it is expected that some platforms will benefit from specific adjustments.
For such case, it's possible to build LZ4 like this : 
`CFLAGS="-O3 -march=prescott" make install`

This would basically perform the same as your patch, but remain in the domain of "system specific" settings.